src/Models/HydrostaticFreeSurfaceModels/HydrostaticFreeSurfaceModels.jl

```diff
@@ -75,9 +75,19 @@ Return a flattened `NamedTuple` of the fields in `model.velocities`, `model.free
 `model.tracers`, and any auxiliary fields for a `HydrostaticFreeSurfaceModel` model.
 """
 @inline fields(model::HydrostaticFreeSurfaceModel) = 
-    merge(hydrostatic_fields(model.velocities, model.free_surface, model.tracers),
-          model.auxiliary_fields,
-          biogeochemical_auxiliary_fields(model.biogeochemistry))
+    merge(hydrostatic_fields(model.velocities, model.free_surface, model.tracers), model.auxiliary_fields)
+
+"""
+    field_names(model)
+
+Return a tuple of the field names associated with `HydrostaticFreeSurfaceModel`. 
+Equivalent to doing `keys(fields(model))`.
+"""
+@inline field_names(model::HydrostaticFreeSurfaceModel) = field_names(model.free_surface, tracernames(model.tracers), model.auxiliary_fields)
+
+@inline field_names(free_surface, tracernames, auxiliary_fields) = (:u, :v, :w, tracernames..., :η, keys(auxiliary_fields)...)
+@inline field_names(::Nothing, tracernames, auxiliary_fields)    = (:u, :v, :w, tracernames..., keys(auxiliary_fields)...)
+@inline field_names(::SplitExplicitFreeSurface, tracernames, auxiliary_fields) = (:u, :v, :w, tracernames..., :η, :U, :V, keys(auxiliary_fields)...)
```

Comment:
I think this is a bad idea. I think that `fields` and `prognostic_fields` should be the source of truth. Those functions return NamedTuple --- both names, and fields. Thus the names can be derived from `fields` or `prognostic_fields`.

Reply:
we need something to regularize the boundary conditions, so we need some names that correspond to `keys(fields(model))` before the model is built. I am open to a different implementation though.

Reply:
I see

Reply:
I have changed the design for the `fields(model)` to always output a predictable namedtuple with `nothing` in place of fields that don't exist. In this way we don't need a new function